tests/test_app/test_data_store_new.py

```diff
@@ -65,27 +69,12 @@ def nc_dir(tmp_dir):
     logs_dir = nc_dir / _LOG_TABLE
     not_dir = nc_dir / _NOT_COMPLETED_TABLE
     md5_dir = nc_dir / _MD5_TABLE
-    for fn in list(logs_dir.glob("*.log")):
-        fn.unlink()
-    for fn in list(not_dir.glob("*.json")):
-        fn.unlink()
-    for fn in list(md5_dir.glob("*.txt")):
-        fn.unlink()
-    for fn in list(nc_dir.glob("*.*")):
-        fn.unlink()
-    if Path(logs_dir).exists():
-        Path(logs_dir).rmdir()
-    if Path(not_dir).exists():
-        Path(not_dir).rmdir()
-    if Path(md5_dir).exists():
-        Path(md5_dir).rmdir()
-    if Path(nc_dir).exists():
-        Path(nc_dir).rmdir()
     nc_dir.mkdir(parents=True, exist_ok=True)
     logs_dir.mkdir(exist_ok=True)
     not_dir.mkdir(exist_ok=True)
     (logs_dir / "scitrack.log").write_text((DATA_DIR / "scitrack.log").read_text())
-    return nc_dir
+    yield nc_dir
+    shutil.rmtree(nc_dir)
 
```

Comment:
I have some small changes. I think it is better to put it in a commit instead of explaining all here. If you merge this, I will push it.